xdsl/irdl/constraints.py

```diff
@@ -855,6 +984,14 @@ def infer(
         attr = self.constr.infer(context)
         return (attr,) * length
 
+    def mapping_type_vars(
+        self, type_var_mapping: dict[TypeVar, AttrConstraint]
+    ) -> GenericRangeConstraint[AttributeCovT]:
+        constr = self.constr.mapping_type_vars(type_var_mapping)
+        if constr is self.constr:
+            return self
+        return RangeOf(constr, length=self.length)
```

Comment:
Either int constraints shouldn't have a `mapping_type_vars` or we should use it here.